enumerable.rb

```diff
@@ -0,0 +1,119 @@
+# Enumerable method
+module Enumerable
+  def my_each
+    arr = self
+    if arr.is_a?(Array)
+      (0..(arr.length - 1)).each do |i|
+        yield(arr[i])
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index
+    arr = self
+    (0..(arr.length - 1)).each do |i|
+      yield(arr[i], i)
+    end
+    arr
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    puts result.to_s
+    result
+  end
+
+  def my_all?
+    my_each do |num|
+      if yield(num) == false
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
+
+  def my_any?
+    my_each do |num|
+      if yield(num) == true
+        puts true
+        return true
+      end
+    end
+    puts false
+    false
+  end
```

Comment:
- #my_any? returns false if the block ever returns a value false or nil in a range
- #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- Should work as #any? when a Regex, Class or pattern is given as an argument